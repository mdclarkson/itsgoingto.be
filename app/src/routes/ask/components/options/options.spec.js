import React from 'react'
import { render, screen } from '@testing-library/react'
import Options from './options'

const defaultProps = {
  hasQuestion: true,
  poll: { question: 'Question', answers: [] },
  updateOptions: () => {}
}

describe('(Route) Ask', () => {
  describe('(Component) Options', () => {
    describe('(Render)', () => {
      describe('when hasQuestion is false', () => {
        it('options has class gone', () => {
          render(<Options {...defaultProps} hasQuestion={false} />)

          const options = screen.getByTestId('options')
          expect(options).toHaveClass('gone')
        })
      })

      describe('when hasQuestion is true', () => {
        it('options does not have class gone', () => {
          render(<Options {...defaultProps} hasQuestion />)

          const options = screen.getByTestId('options')
          expect(options).not.toHaveClass('gone')
        })
      })
      it('matches snapshot', () => {
        const { asFragment } = render(<Options {...defaultProps} />)

        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
})
