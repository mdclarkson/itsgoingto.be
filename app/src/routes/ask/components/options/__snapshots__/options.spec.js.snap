// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Route) Ask (Component) Options (Render) matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="options"
    data-testid="options"
  >
    <div
      class="input-option input-option-multipleChoice"
    >
      <input
        class="input-checkbox input-checkbox-multipleChoice"
        id="multiple-choice"
        name="multiple"
        type="checkbox"
      />
      <label
        class="input-label input-label-options input-label-multipleChoice"
        for="multiple-choice"
      >
        Multiple choice responses
      </label>
    </div>
    <div
      class="input-option input-option-passphrase"
    >
      <label
        class="input-label input-label-passphrase"
        for="passphrase"
      >
        Set a passphrase (optional)
        <input
          class="input-field input-field-passphrase"
          id="passphrase"
          name="passphrase"
          type="text"
          value=""
        />
      </label>
    </div>
    <div
      class="input-option input-option-endPoll"
    >
      <div
        class="input-option-radios"
      >
        <input
          checked=""
          class="input-radio input-checkbox-endType"
          id="end-never"
          name="endType"
          type="checkbox"
          value="endNever"
        />
        <label
          class="input-label input-label-options input-label-endType"
          for="end-never"
        >
          Don't End Poll
        </label>
        <input
          class="input-radio input-checkbox-endType"
          id="end-at"
          name="endType"
          type="checkbox"
          value="endAt"
        />
        <label
          class="input-label input-label-options input-label-endType"
          for="end-at"
        >
          End Poll At
        </label>
        <input
          class="input-radio input-checkbox-endType"
          id="end-in"
          name="endType"
          type="radio"
          value="endIn"
        />
        <label
          class="input-label input-label-options input-label-endType"
          for="end-in"
        >
          End Poll In
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
